automation/taskcluster/decision_task_snapshot_release.py

```diff
@@ -0,0 +1,184 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+"""
+Decision task for triggering a snapshot release
+"""
+
+from __future__ import print_function
+import json
+import os
+import taskcluster
+import sys
+
+import lib.module_definitions
+import lib.tasks
+import lib.util
+
+
+TASK_ID = os.environ.get('TASK_ID')
+REPO_URL = os.environ.get('MOBILE_HEAD_REPOSITORY')
+BRANCH = os.environ.get('MOBILE_HEAD_BRANCH')
+HEAD_REV = os.environ.get('MOBILE_HEAD_REV')
+
+BUILDER = lib.tasks.TaskBuilder(
+    task_id=TASK_ID,
+    repo_url=os.environ.get('MOBILE_HEAD_REPOSITORY'),
+    branch=os.environ.get('MOBILE_HEAD_BRANCH'),
+    commit=HEAD_REV,
+    owner="skaspari@mozilla.com",
+    source='https://github.com/mozilla-mobile/android-components/raw/{}/.taskcluster.yml'.format(HEAD_REV),
+    scheduler_id=os.environ.get('SCHEDULER_ID'),
+)
+
+
+def generate_build_task(info):
+    checkout = ("export TERM=dumb && git fetch {} {} && "
+                "git config advice.detachedHead false && "
+                "git checkout {} ".format(REPO_URL, BRANCH, HEAD_REV))
+    artifacts = {
+        info["artifact"]: {
+            'type': 'file',
+            'expires': taskcluster.stringDate(taskcluster.fromNow('1 year')),
+            'path': info['path']
+        }
+    }
+    module_name = ":{}".format(info['name'])
+
+    return taskcluster.slugId(), BUILDER.build_task(
+        name='Android Components - Module {}'.format(module_name),
+        description='Building and testing module {}'.format(module_name),
+        command=(checkout +
+                 ' && ./gradlew --no-daemon clean ' +
+                 "-Pcoverage -Psnapshot " +
+                 " ".join(["{}:{}".format(module_name, gradle_task) for gradle_task in ['assemble', 'test', 'lint']])  +
```

Comment:
We should build the components without `-Pcoverage` since we do not want to measure code coverage here. And as far as I know that instruments the code and makes it slower and that's not what we want to release.